tests/components/snapcast/test_media_player.py

```diff
@@ -0,0 +1,162 @@
+"""Test the snapcast media player implementation."""
+
+from typing import Any
+from unittest.mock import patch
+
+from snapcast.control.server import CONTROL_PORT
+
+from homeassistant.components.media_player import (
+    ATTR_INPUT_SOURCE,
+    ATTR_INPUT_SOURCE_LIST,
+    ATTR_MEDIA_ALBUM_ARTIST,
+    ATTR_MEDIA_ALBUM_NAME,
+    ATTR_MEDIA_ARTIST,
+    ATTR_MEDIA_DURATION,
+    ATTR_MEDIA_POSITION,
+    ATTR_MEDIA_TITLE,
+    ATTR_MEDIA_TRACK,
+    ATTR_MEDIA_VOLUME_LEVEL,
+    ATTR_MEDIA_VOLUME_MUTED,
+    DOMAIN as MEDIA_PLAYER_DOMAIN,
+    SERVICE_SELECT_SOURCE,
+)
+from homeassistant.components.snapcast.const import DOMAIN
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import CONF_HOST, CONF_PORT, STATE_PLAYING
+from homeassistant.core import HomeAssistant
+
+from .const import TEST_CLIENT_ENTITY_ID, TEST_GROUP_ENTITY_ID, TEST_STATE
+
+from tests.common import MockConfigEntry
+
+
+async def _call_service(hass: HomeAssistant, name: str, data: dict[str, Any]) -> None:
+    await hass.services.async_call(
+        MEDIA_PLAYER_DOMAIN, name, service_data=data, blocking=True
+    )
```

Comment:
I personally wouldn't wrap this as it's clearer without wrapper

Reply:
Done